clippy_lints/src/manual_unwrap_or_default.rs

```diff
@@ -53,6 +53,7 @@ declare_lint_pass!(ManualUnwrapOrDefault => [MANUAL_UNWRAP_OR_DEFAULT]);
 
 fn get_some<'tcx>(cx: &LateContext<'tcx>, pat: &Pat<'tcx>) -> Option<HirId> {
     if let PatKind::TupleStruct(QPath::Resolved(_, path), &[pat], _) = pat.kind
+        && matches!(pat.kind, PatKind::Binding(..))
```

Comment:
If we make this `&& let PatKind::Binding(....)` to get the ID we can return that directly instead of https://github.com/rust-lang/rust-clippy/pull/12961/files#diff-4b97f32b70a6481322fccdf7cc1c61f815ba832fd5e961607e2f708c42d83ffcR64-R70